src/main/java/com/weirddev/testme/intellij/builder/MethodFactory.java

```diff
@@ -169,6 +170,32 @@ private static List<Param> extractMethodParams(PsiMethod psiMethod, boolean shou
         return params;
     }
 
+
+    //analyze the exception types of the method
+    private static String extractMethodExceptionTypes(PsiMethod psiMethod ,Boolean throwSpecificExceptionTypes,boolean testable) {
+        if(!testable){
+            return null;
+        }
+        String throwsExceptions = null;
+        if(throwSpecificExceptionTypes){
+            PsiReferenceList throwsList = psiMethod.getThrowsList();
+            PsiClassType[] referencedTypes = throwsList.getReferencedTypes();
+
+            for (PsiClassType type : referencedTypes) {
```

Comment:
can be simplified. please replace with:
```java
String throwsExceptions = Arrays.stream(referencedTypes)
    .map(PsiClassType::resolve)
    .filter(Objects::nonNull)
    .map(PsiClass::getQualifiedName)
    .filter(Objects::nonNull)
    .collect(Collectors.joining(","));
return throwsException.isEmpty() ? "Exception" : throwsException;
```

Reply:
thanks，I will Modify as per your advice and add  integration test💪